braindecode/models/conformer.py

```diff
@@ -0,0 +1,279 @@
+
+    Parameters PatchEmbedding
+    -------------------------
+    - n_kernel: int
+        Number of kernels for the temporal convolution layer (first layer).
+    - n_kernel_temp_conv: int
+        Kernel size of the temporal convolution layer.
+    - eeg_channel: int
+        Number of EEG channels (kernel size of the spatial convolution layer).
+    - n_kernel_avg_pool: int
+        Kernel size of the average pooling layer.
+    - stride_avg_pool: int
+        Stride of the average pooling layer.
+    - drop_prob: float
+        Dropout rate of the convolutional layer.
+
+    Parameters TransformerEncoder
+    -----------------------------
+    - att_depth: int
+        Number of self-attention layers.
+    - att_heads: int
+        Number of attention heads.
+    - att_drop: float
+        Dropout rate of the self-attention layer.
+
+    Parameters ClassificationHead
+    -----------------------------
+    - fc_dim: int
+        The dimension of the fully connected layer.
+    - n_classes: int
+        Number of classes.
+
+
+    References
+    ----------
+    .. [EEG Conformer] Song, Y., Zheng, Q., Liu, B. and Gao, X., 2022. EEG
+       conformer: Convolutional transformer for EEG decoding and visualization.
+       IEEE Transactions on Neural Systems and Rehabilitation Engineering,
+       31, pp.710-719. https://ieeexplore.ieee.org/document/9991178
+    .. [EEG Conformer Code] Song, Y., Zheng, Q., Liu, B. and Gao, X., 2022. EEG
+       conformer: Convolutional transformer for EEG decoding and visualization.
+       https://github.com/eeyhsong/EEG-Conformer.
+
+
+    """
+
+    def __init__(
+            self,
+            kernel=40,
+            kernel_temp_conv=25,
+            eeg_channel=22,
+            kernel_avg_pool=75,
+            stride_avg_pool=15,
+            conv_drop=0.5,
+            att_depth=6,
+            att_heads=10,
+            att_drop=0.5,
+            fc_dim=2440,
+            n_classes=4,
+    ):
+        super().__init__(
+            PatchEmbedding(
+                kernel,
+                kernel_temp_conv,
+                eeg_channel,
+                kernel_avg_pool,
+                stride_avg_pool,
+                conv_drop,
+            ),
+            TransformerEncoder(att_depth, kernel, att_heads, att_drop),
+            ClassificationHead(kernel, fc_dim, n_classes),
+        )
+
+
+class PatchEmbedding(nn.Module):
+    """Patch Embedding.
+
+    The authors used a convolution moduleto capture local features,
+    instead of postion embedding.
+
+    """
+
+    def __init__(
+            self,
+            kernel,
+            kernel_temp_conv,
+            eeg_channel,
+            kernel_avg_pool,
+            stride_avg_pool,
+            conv_drop,
+    ):
+        super().__init__()
+
+        self.shallownet = nn.Sequential(
+            nn.Conv2d(1, kernel, (1, kernel_temp_conv), (1, 1)),
+            nn.Conv2d(kernel, kernel, (eeg_channel, 1), (1, 1)),
```

Comment:
Defining the stride is not necessary as 1 is the default.
```suggestion
            nn.Conv2d(1, kernel, (1, kernel_temp_conv)),
            nn.Conv2d(kernel, kernel, (eeg_channel, 1)),
```

Reply:
Okay. Thanks for your help!